### ISSUE TITLE
Sent E-Mails are not being fetched and converted

### ISSUE THREAD

**Issue by pixi** [creecros/Mailmagik#31]
Unfortunately after setting this up in my installation on a Shared hoster E-Mails didn't make it into the Project.

What I tried so far (w/o success):
• Port set to 143 or empty
• Port set to empty
• Advanced settings set to /readonly
• Advanced settings set to empty (will be replaced by /imap/ssl after "Save")
• Username set to E-Mail address as well as serverinternal Username 
• Selected "Parse from the "SUBJECT" field" and send emails with appropriate Prefix
• Selected "Parse from the "TO" field" and send emails with appropriate Prefix
• used the CLI on the server to "php cli mailmagik:fetchmail"

Automatic action is set to:
• Automatically Convert Emails to Tasks
Event name = Trigger Mailmagik mail fetching
Column = To be done (An existing column)
Color = Red

QUESTION:
1) Is there a way I can debug w/o having access to server logs
2) What else can be changed to narrow down the issue?
3) How else can I test if Mailmagik can "see" the content in an IMAP folder?

I'm looking for ages for such a feature! Thank you for providing it at all.

P.S. It's the latest Kanboard and all Plugins are up-to-date.

**Comment by creecros** [comment]
I have no experience trying this with a shared hoster email(not 100% sure what that is to be honest), maybe @alfredbuehler or @aljawaid have.

First, make sure you satisfy all the prerequisites listed in the Readme. That's very important.  And if you don't have access to the server logs, my first guess would be you don't have access to install the prerequisites, so double check that, first.

**Comment by pixi** [comment]
According to the ISP installing PHP IMAP is not possible.
Is there any other way to make this work? Would a cron the only other chance? And if so, what commands are required?

P.S. A more detailed and ordered list of requirements [here](https://github.com/creecros/Mailmagik#requirements ) would have helped (me) not to waste your time.

**Comment by creecros** [comment]
No, unfortunately, it will not function without php-imap.

**Comment by pixi** [comment]
It's all-inkl.com
They even have Kanboard as Software to install automatically. That's why I think it's really a pity.

**Comment by creecros** [comment]
Crazy, their website says "many installable modules" for php. I'd complain!

**Comment by alfredbuehler** [comment]
There are several hosting packages available at all-inkl.com.  Maybe you need an upgrade.

**Comment by pixi** [comment]
Yes, I did complain! I wrote and called them twice and they just said: "Check the modules using a phpinfo.php. If you need more you can rent a Managed Server for 99,- /month". 
As much as I like Kanboard and would love to have any kind of Mail-to-Task that is by far too much!

That is what the info gave me reg. imap:
![ai_imap](https://github.com/creecros/Mailmagik/assets/7418913/c4e5467e-d455-46c5-b409-31e95ee12bd2)

**Comment by alfredbuehler** [comment]
But then it should work.

**Comment by alfredbuehler** [comment]
@pixi Any news here? I'd be ready to help/assist you.

**Comment by pixi** [comment]
@alfredbuehler unfortunately not. I'm a bit stuck 'cos I'm not tec savvy in everything Web.

Is there any way to test/log what's wrong? Or is there any other way to figure that out?

I'd even pay you (if that is of any interest ;o)

**Comment by alfredbuehler** [comment]
@pixi 
> Is there any way to test/log what's wrong? Or is there any other way to figure that out?

Not that I'm aware of.

Please start with these settings.
![image](https://github.com/creecros/Mailmagik/assets/2079289/073d1364-4c9b-4df9-95f3-8def49f650b9)
- Send a mail to one of your tasks, eg "Task#123" <yourmail\@youdomain>.
- Got to that task, select "Task email" ins the side panel.
- If something is wrong, after the timeout, a error message in shown at the top.
![image](https://github.com/creecros/Mailmagik/assets/2079289/d57f6946-ddd8-4efe-a3d6-c962838ffdf5)

Two things:
- It would be helpful if you could provide more information about the IMAP config.
- If you prefer to continue in German, in a chat room, you can pass me your data **via my e-mail**. I have IRC, XMPP, matrix, Threema.

**Comment by pixi** [comment]
Okay, that worked so far. After refreshing "Task Email" I saw the Email.

What information about the IMAP config do you still need?

Yes, it would be useful to communicate directly. I still have matrix and get back on tomorrow, WED. 12.

**Comment by creecros** [comment]
If you saw the email, it works. Nothing is wrong with your imap config.

**Comment by alfredbuehler** [comment]
@pixi 
> What information about the IMAP config do you still need?

No, as your connection is working now.

**Comment by pixi** [comment]
Problem is solved thanks to @alfredbuehler 

Some remarks:
• There was a Sympfony issue which got solved in Mailmagik, 1.2.3 when using KB 1.2.32. Use this or later versions!
• The external E-mail send to KB **MUST** come from an E-mail address of a member of the Project! Just an external E-mail address does not work.
• To use cron on a shared hoster I added a 'cron.php' to the KB directory in a seprate dir.:
```
<?php
exec("/bin/bash /www/htdocs/fullpathto/kanboard/fetchmail_mailmagik/fetchmail.sh) 2>&1", $out, $result);
echo "Returncode: " .$result ."<br>";
echo "Ausgabe des Scripts: " ."<br>";
echo "<pre>"; print_r($out);
?>
```
 and a 'fetchmail.sh' in the same

`/usr/bin/php81 -f /www/htdocs/fullpathto/kanboard/fetchmail_mailmagik/cli mailmagik:fetchmail`

and pointed the hosters cron setting to the php.

**Comment by alfredbuehler** [comment]
> The external E-mail send to KB **MUST** come from an E-mail address of a member of the Project! 

I've added a remark to the README in v1.2.3.
